Lib/test/datetimetester.py

```diff
@@ -6781,6 +6782,47 @@ def test_datetime_from_timestamp(self):
                     self.assertEqual(dt_orig, dt_rt)
 
 
+class ExtensionModuleTests(unittest.TestCase):
+
+    def setUp(self):
+        if self.__class__.__name__.endswith('Pure'):
+            self.skipTest('Not relevant in pure Python')
+
+    def test_gh_120161(self):
+        with self.subTest('simple'):
+            script = textwrap.dedent("""
+                import datetime
+                from _ast import Tuple
+                f = lambda: None
+                Tuple.dims = property(f, f)
+
+                class tzutc(datetime.tzinfo):
+                    pass
+                """)
+            script_helper.assert_python_ok('-c', script)
+
+        with self.subTest('complex'):
+            script = textwrap.dedent("""
+                import asyncio
```

Comment:
I would separate this test into its own method, it imports `asyncio`. Which means that it needs `@requires_working_socket()` decorator

Reply:
Btw, why does it need `asyncio`? 🤔

Reply:
Because of the original issue (the snippet is taken almost verbatim).

Reply:
Importing `asyncio` on its own is not enough to be a problem.  I was able to verify that it works fine on a WASI build (where `@requires_working_socket()` would skip).  This is also verified by the WASI CI check passing.  Unless there is another reason to do so, I'd rather not add that decorator.